### PR TITLE
Pattern Explorer Modal: Select the first category as the initial category

### LINKED ISSUE #47657: Pattern Explorer should open up on the "Featured" category
### Description

Originally reported as part of https://github.com/WordPress/gutenberg/issues/47582, the Pattern Explorer should start with the "Featured" category, instead of with no category selected. 

### Step-by-step reproduction instructions

1. Open the Pattern Inserter.
2. Click the "Explore all patterns" button.
3. See Pattern Explorer open with no category selected, thus no pattern results.

#### Current: 
CleanShot 2023-02-01 at 14 16 42

#### Proposed (opened on featured category): 
CleanShot 2023-02-01 at 14 17 40

### Environment info

- WordPress 6.1.1
- Gutenberg trunk

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by apeatling**
Fixes #47657

## What?

If no inserter sidebar pattern category has been selected, make sure the first pattern category is selected as the initially selected category in the pattern explorer modal.

## Why?

This seems to be a regression, where there's no pattern category selected when the modal first opens so the whole thing is empty. This is only if no pattern category has been selected in the inserter sidebar.

## How?

Take the first category object in the `categories` array and pass that as `initialCategory`. 

## Testing Instructions

1. Open the post or site editor.
2. Open the inserter sidebar
3. Select "Explore all patterns" button at the bottom of the sidebar
4. Confirm the first pattern category is selected when the modal opens.
5. Close the modal and select a pattern category in the inserter sidebar.
6. Select the "Explore all patterns" button again, and confirm that the same pattern category is selected in the modal.

## Screenshots or screencast 
**Before**:
Screenshot 2023-02-01 at 4 06 21 PM

**After:**
Screenshot 2023-02-01 at 4 05 47 PM

### CODE REVIEW

File: packages/block-editor/src/components/inserter/block-patterns-tab.js [hunk 1]
```diff
@@ -195,6 +195,8 @@ function BlockPatternsTabs( {
 } ) {
 	const [ showPatternsExplorer, setShowPatternsExplorer ] = useState( false );
 	const categories = usePatternsCategories( rootClientId );
+	const initialCategory =
+		selectedCategory || ( categories.length && categories[ 0 ] );
```
Comment (Mamaduka) [inline packages/block-editor/src/components/inserter/block-patterns-tab.js]:
```suggestion
	const initialCategory = selectedCategory || categories[ 0 ];
```

Suggestion (non-blocking): I think it's safe to assume that `categories` returned from `usePatternsCategories` always will be an array.

**Review by Mamaduka — APPROVED** [review]
It works as expected ✅

PR will require a rebase to restart Playwright tests. See #47629.

**Review by carolinan — APPROVED** [review]
I have followed the testing instructions and the PR solves the problem 👍

File: packages/block-editor/src/components/inserter/block-patterns-tab.js (hunk 1 above)
Comment (apeatling) [reply packages/block-editor/src/components/inserter/block-patterns-tab.js]: Sounds good -- are you sure the array will always have at least one item?

Comment (Mamaduka) [reply packages/block-editor/src/components/inserter/block-patterns-tab.js]: > Are you sure the array will always have at least one item?

I think users can unregister any or all categories. But empty array fine.

Following will return `undefined`.
```
const categories = [];
categories[0]
```

**Comment by apeatling** [comment]
Thanks for the reviews! :+1:

**Comment by Mamaduka** [comment]
I just cherry-picked this PR to the release/15.1 branch to get it included in the next release: 8a15cebb9f

**Pull request merged**